pkg/fqdn/matchpattern/matchpattern.go

```diff
@@ -68,5 +69,9 @@ func ToRegexp(pattern string) string {
 	pattern = strings.Replace(pattern, ".", "[.]", -1)
 
 	// Anchor the match to require the whole string to match this expression
-	return "^" + pattern + "$"
+	pattern = "^" + pattern + "$"
+	if origPattern == "*" {
+		pattern = "(" + pattern + ")|(^[.]$)"
```

Comment:
Did you consider changing the line 60 instead, to be like:
```
        if pattern == "*" {
		return "(^(" + allowedDNSCharsREGroup + "+[.])+$)|(^[.]$)"
	}
```
Then no other changes would be needed?

Reply:
I toyed with that line for a bit but didn't consider the early exit approach. I agree this would be simpler.